tests/utils/linear_feedback_controller.hpp

```diff
@@ -0,0 +1,113 @@
+#ifndef LINEAR_FEEDBACK_CONTROLLER_TESTS__LINEAR_FEEDBACK_CONTROLLER_HPP_
+#define LINEAR_FEEDBACK_CONTROLLER_TESTS__LINEAR_FEEDBACK_CONTROLLER_HPP_
+
+#include <chrono>  // duration_cast/milliseconds
+
+#include "linear_feedback_controller/linear_feedback_controller.hpp"
+#include "pd_controller.hpp"
+#include "robot_model.hpp"
+
+namespace tests::utils {
+
+inline auto MakeAllControllerParametersFrom(
+    std::string_view urdf,
+    std::initializer_list<std::vector<JointDescription>> all_joint_lists,
+    std::initializer_list<linear_feedback_controller::Duration> durations)
+    -> std::vector<linear_feedback_controller::ControllerParameters> {
+  using linear_feedback_controller::ControllerParameters;
+
+  const auto FromEigen = [](const Eigen::VectorXd& val) -> std::vector<double> {
+    std::vector<double> out;
+    out.resize(val.size());
+    Eigen::Map<Eigen::VectorXd>(out.data(), out.size()) = val;
+    return out;
+  };
+
+  std::vector<ControllerParameters> out;
+
+  out.reserve(2 * all_joint_lists.size() * durations.size());
+  for (const auto& joint_list : all_joint_lists) {
+    const auto [controlled, moving] = JointNamesPair::From(joint_list);
+
+    // FIXME : Controlled or moving size ?
```

Comment:
We need gains for controlled joints. Moving joints only tell us the state of the robot. This is only equal for manipulators.